cli/tauri.js/src/api/recipes/react.ts

```diff
@@ -0,0 +1,58 @@
+import { Recipe } from '.'
+import { TauriBuildConfig } from '../../types/config'
+import { spawnSync } from '../../helpers/spawn'
+import logger from '../../helpers/logger'
+import copyTemplates from '../../helpers/copy-templates'
+import { resolve, join } from 'path'
+
+const uiAppDir = 'app-ui'
+
+const log = logger('react-recipe')
+
+const completeLogMsg = `
+  Your installation completed.
+  To start, run yarn tauri dev
+`
+
+const afterCra = (): void => {
+  copyTemplates({
+    source: resolve(__dirname, '../../templates/recipes/react/'),
+    scope: {},
+    target: join(uiAppDir, './src/')
+  })
+  log(completeLogMsg)
+}
+
+const reactjs: Recipe = {
+  descriptiveName: 'React.js',
+  shortName: 'reactjs',
+  configUpdate: (cfg: TauriBuildConfig): TauriBuildConfig => ({
+    ...cfg,
+    distDir: `../${uiAppDir}/build`,
+    devPath: 'http://localhost:3000',
+    beforeDevCommand: `yarn --cwd ${uiAppDir} start`,
```

Comment:
I think we can specify a `BROWSER` env var here so it doesn't open the browser and the Tauri app.
https://create-react-app.dev/docs/advanced-configuration

Reply:
22bcf3a

Reply:
Nevermind, I reverted that as it's not going to work on windows. beforeDevCommand is executed using `exec` from `process`, which does spawn a shell and tries to run it.

So implementing this would require a change to the build config to allow beforeDevCommand to specify an environment variable, 

Something like 

```
interface Command {
  env: {[name: string]: string},
  command: string
}
interface TauriBuildConfig {
   beforeDevCommand: string | Command
}
```

This is not a big deal, but I think it lets the scope of this PR creep a little bit too much, I would prefer to do this in a follow-on PR if that's ok.

Reply:
aye, agreed

Reply:
Also, environment variables are passed into the child process, so for example `BROWSER=none yarn tauri dev` works on linux/osx,